app/src/main/kotlin/compiler/intermediate_form/FunctionDependenciesAnalyzer.kt

```diff
@@ -24,18 +24,39 @@ object FunctionDependenciesAnalyzer {
             return uniqueIdentifiers[function]!!.value
         }
 
-        fun analyze(node: Any, pathSoFar: String?) {
+        fun analyze(node: Any, pathSoFar: String? = null) {
             when (node) {
                 is Program.Global.FunctionDefinition -> { analyze(node.function, pathSoFar) }
                 is Statement.FunctionDefinition -> { analyze(node.function, pathSoFar) }
                 is Function -> {
                     val newPrefix = nameFunction(node, pathSoFar)
-                    node.body.forEach { analyze(it, newPrefix) }
+                    var blockNumber = -1
+                    for (statement in node.body) {
+                        when (statement) {
+                            is Statement.Block -> {
+                                blockNumber++
+                                statement.block.forEach { analyze(it, newPrefix + blockNumber) }
+                                analyze(statement, newPrefix + blockNumber)
```

Comment:
I don't understand what this (and line 50) is for.

Reply:
Leftover from copying, not necessary